private/pkg/git/repository.go

```diff
@@ -333,12 +329,17 @@ func (r *repository) readPackedRefs() error {
 	return r.packedReadError
 }
 
+// detectDefaultBranch returns the repository's default branch name. It attempts to read it from the
+// `.git/refs/remotes/origin/HEAD` file, and expects it to be pointing to a branch also in the
+// `origin` remote.
 func detectDefaultBranch(gitDirPath string) (string, error) {
+	const defaultRemoteName = "origin"
```

Comment:
This is the only place in which we keep assuming `origin` remote name. If no explicit default branch is passed, then we read the HEAD of the default remote name (`origin`).

Reply:
This is reasonable to me -- this is more a note for myself, but I think at some point, down the road, we may need to expose this so that it can be used by the BSR in some way, but this is a reasonable way of handling this right now.